engine/src/flutter/impeller/entity/geometry/rect_geometry.cc

```diff
@@ -207,35 +339,17 @@ std::optional<Rect> StrokeRectGeometry::GetCoverage(
   return rect_.TransformBounds(transform);
 }
 
+GeometryResult::Mode StrokeRectGeometry::GetResultMode() const {
+  // Most(all?) of our geometry is produced in full with an explicit mode
+  // specified in the returned GeometryResult, but just in case we missed
+  // a case, the safest mode to return here is kPreventOverdraw.
+  return GeometryResult::Mode::kPreventOverdraw;
```

Comment:
I may be missing something obvious. But this should no longer be necessary now right?

Reply:
It also runs 50% slower.

Reply:
kPreventOverdraw isn't magic. It costs - a lot.

Reply:
Yeah, so why specify that as the default on the off chance we missed some edge case.

Reply:
It's not just the most common case, it never falls through. I removed it for now, but I don't like the entire design of this API that assumes things that have consequences that aren't immediately obvious and we construct an entire operation.